bindings/netstandard/ElectionGuard/ElectionGuard.ElectionSetup/ElectionPolynomial.cs

```diff
@@ -13,78 +13,162 @@ namespace ElectionGuard.ElectionSetup;
+        for (var i = 1; i < numberOfCoefficients; i++)
+        {
+            Coefficients.Add(new Coefficient());
+        }
+    }
 
-        for (var i = 0; i < numberOfCoefficients; i++)
+    /// <summary>
+    /// Generates a polynomial for sharing election keys using the provided key pair as the zero-index coefficient.
+    /// Each coefficient is an exponential order for the polynomial and the guardian secret key is the 0-index coefficient.
+    /// </summary>
+    /// <param name="numberOfCoefficients">Number of coefficients of polynomial, typically the quorum count of guardians</param>
+    public ElectionPolynomial(
+        [Range(1, int.MaxValue)] int numberOfCoefficients, ElGamalKeyPair keyPair)
+    {
+        Coefficients = new List<Coefficient> { new(keyPair.SecretKey) };
+        for (var i = 1; i < numberOfCoefficients; i++)
         {
-            coefficients.Add(GenerateCoefficient(nonce, i));
+            Coefficients.Add(new Coefficient());
         }
+    }
 
-        return new(coefficients);
+    /// <summary>
+    /// Generates a polynomial for sharing election keys.
+    /// Each coefficient is an exponential order for the polynomial and the guardian secret key is the 0-index coefficient.
+    /// </summary>
+    public ElectionPolynomial(List<Coefficient> coefficients)
+    {
+        if (coefficients.Any(i => !i.IsValid()))
+        {
+            throw new ArgumentException("Invalid coefficients provided");
+        }
+        Coefficients = coefficients;
     }
 
-    private static Coefficient GenerateCoefficient(ElementModQ? nonce, int i)
+    /// <summary>
+    /// Compute the single coordinate value of the polynomial at a given degree for use in sharing election keys.
+    /// </summary>
+    /// <param name="degree">The exponential degree of the polynomial (usually the sequence order)</param>
+    public ElementModQ ComputeCoordinate(ulong degree)
     {
-        var secretKey = GenerateSecretKey(nonce, i);
-        var keypair = ElGamalKeyPair.FromSecret(secretKey);
-        var seed = BigMath.RandQ();
-        SchnorrProof proof = new(keypair, seed);
+        return ComputeCoordinate(new ElementModQ(degree));
+    }
 
-        return new(keypair, proof);
+    /// <summary>
+    /// Compute the single coordinate value of the polynomial at a given degree for use in sharing election keys.
+    /// </summary>
+    /// <param name="degree">The exponential degree of the polynomial (usually the sequence order)</param>
+    public ElementModQ ComputeCoordinate(ElementModQ degree)
+    {
+        var computedValue = Constants.ZERO_MOD_Q; // start at 0 mod q.
+
+        foreach (var (coefficient, index) in Coefficients.WithIndex())
+        {
+            using var exponent = BigMath.PowModQ(degree, index);
+            using var factor = BigMath.MultModQ(coefficient.Value, exponent);
+
+            computedValue = BigMath.AddModQ(computedValue, factor);
```

Comment:
Might want to look at the new function for ElementModP for MultModP that does not require the first parameter to  constantly be disposed and making a new version.

Reply:
https://github.com/microsoft/electionguard-core2/issues/189